features/app_bundle.feature

```diff
@@ -3,71 +3,77 @@ Feature: Generating Android app bundles
-    And the payload field "appVersionCode" equals "1" for request 2
+    And the request is valid for the Android Mapping API
+    And the field "apiKey" for multipart request equals "TEST_API_KEY"
+    And the field "versionCode" for multipart request equals "1"
+    And the field "versionName" for multipart request equals "1.0"
+    And the field "appId" for multipart request equals "com.bugsnag.android.example.bar"
+    And I discard the oldest request
 
-    And the request 1 is valid for the Android Mapping API
-    And the field "apiKey" for multipart request 1 equals "TEST_API_KEY"
-    And the field "versionCode" for multipart request 1 equals "1"
-    And the field "versionName" for multipart request 1 equals "1.0"
-    And the field "appId" for multipart request 1 equals "com.bugsnag.android.example.bar"
+    And the request is valid for the Build API
+    And the payload field "appVersion" equals "1.0"
+    And the payload field "apiKey" equals "TEST_API_KEY"
+    And the payload field "appVersionCode" equals "1"
+    And I discard the oldest request
 
-    And the request 3 is valid for the Android Mapping API
-    And the field "apiKey" for multipart request 3 equals "TEST_API_KEY"
-    And the field "versionCode" for multipart request 3 equals "1"
-    And the field "versionName" for multipart request 3 equals "1.0"
-    And the field "appId" for multipart request 3 equals "com.bugsnag.android.example.foo"
+    And the request is valid for the Android Mapping API
+    And the field "apiKey" for multipart request equals "TEST_API_KEY"
+    And the field "versionCode" for multipart request equals "1"
+    And the field "versionName" for multipart request equals "1.0"
+    And the field "appId" for multipart request equals "com.bugsnag.android.example.foo"
 
 Scenario: Bundling single flavor
     When I bundle the "Foo" variantOutput for "flavors" using the "standard" bugsnag config
-    Then I should receive 2 requests
+    And I wait to receive 2 requests
 
-    And the request 0 is valid for the Build API
+    Then the request is valid for the Build API
+    And I discard the oldest request
 
-    And the request 1 is valid for the Android Mapping API
-    And the field "apiKey" for multipart request 1 equals "TEST_API_KEY"
-    And the field "versionCode" for multipart request 1 equals "1"
-    And the field "versionName" for multipart request 1 equals "1.0"
-    And the field "appId" for multipart request 1 equals "com.bugsnag.android.example.foo"
+    And the request is valid for the Android Mapping API
+    And the field "apiKey" for multipart request equals "TEST_API_KEY"
+    And the field "versionCode" for multipart request equals "1"
+    And the field "versionName" for multipart request equals "1.0"
+    And the field "appId" for multipart request equals "com.bugsnag.android.example.foo"
 
 Scenario: Auto upload disabled
     When I bundle "default_app" using the "all_disabled" bugsnag config
+    And I wait for 5 seconds
```

Comment:
Is there any way that we can avoid this wait?

Reply:
Not really, we could shorten it to 1-3 seconds, but we need to allow a bit of a wait